silkworm/core/chain/config.cpp

```diff
@@ -124,42 +145,44 @@ std::optional<ChainConfig> ChainConfig::from_json(const nlohmann::json& json) no
-    read_json_config_member(json, "arrowGlacierBlock", config.arrow_glacier_block);
-    read_json_config_member(json, "grayGlacierBlock", config.gray_glacier_block);
+    read_json_config_member(json, "homesteadBlock", config._homestead_block);
+    read_json_config_member(json, "daoForkBlock", config._dao_block);
+    read_json_config_member(json, "eip150Block", config._tangerine_whistle_block);
+    read_json_config_member(json, "eip155Block", config._spurious_dragon_block);
+    read_json_config_member(json, "byzantiumBlock", config._byzantium_block);
+    read_json_config_member(json, "constantinopleBlock", config._constantinople_block);
+    read_json_config_member(json, "petersburgBlock", config._petersburg_block);
+    read_json_config_member(json, "istanbulBlock", config._istanbul_block);
+    read_json_config_member(json, "muirGlacierBlock", config._muir_glacier_block);
+    read_json_config_member(json, "berlinBlock", config._berlin_block);
+    read_json_config_member(json, "londonBlock", config._london_block);
+    read_json_config_member(json, "arrowGlacierBlock", config._arrow_glacier_block);
+    read_json_config_member(json, "grayGlacierBlock", config._gray_glacier_block);
 
     if (json.contains(kTerminalTotalDifficulty)) {
         // We handle terminalTotalDifficulty serialized both as JSON string *and* as JSON number
         if (json[kTerminalTotalDifficulty].is_string()) {
             /* This is still present to maintain compatibility with previous Silkworm format */
-            config.terminal_total_difficulty =
+            config._terminal_total_difficulty =
                 intx::from_string<intx::uint256>(json[kTerminalTotalDifficulty].get<std::string>());
         } else if (json[kTerminalTotalDifficulty].is_number()) {
             /* This is for compatibility with Erigon that uses a JSON number */
             // nlohmann::json treats JSON numbers that overflow 64-bit unsigned integer as floating-point numbers and
             // intx::uint256 cannot currently be constructed from a floating-point number or string in scientific notation
-            config.terminal_total_difficulty =
+            config._terminal_total_difficulty =
                 from_string_sci<intx::uint256>(json[kTerminalTotalDifficulty].dump().c_str());
         }
     }
 
-    read_json_config_member(json, "mergeNetsplitBlock", config.merge_netsplit_block);
-    read_json_config_member(json, "shanghaiTime", config.shanghai_time);
-    read_json_config_member(json, "cancunTime", config.cancun_time);
+    read_json_config_member(json, "mergeNetsplitBlock", config._merge_netsplit_block);
+    read_json_config_member(json, "shanghaiTime", config._shanghai_time);
+    read_json_config_member(json, "cancunTime", config._cancun_time);
 
     /* Note ! genesis_hash is purposely omitted. It must be loaded from db after the
      * effective genesis block has been persisted */
 
+    read_json_config_member(json, "revision", config._revision);
```

Comment:
If we want to modify the genesis json to reflect the fact that we are no longer retrieving the EVMC revision based on block-height or time, we should use the EOSEVM version as a genesis configuration field instead.